org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinker.java

```diff
@@ -208,11 +201,17 @@ protected EObject createProxy(EObject obj, INode node, EReference eRef) {
 		return proxy;
 	}
 
+	private void ensureCacheInitialized() {
```

Comment:
I think we can also mark the LazyLinker itself as an `@Singleton` which would reduce the number of objects, too.

Reply:
This is a good idea. If you do not mind, I would like to do that however in another PR, after Simrel 2025-03 is released, i think making the class a `@Singleton`  could have side-effects if the class or subclasses are not thread-safe.